test/data/surface-phases.yaml

```diff
@@ -157,29 +156,50 @@ graphite-anode-species:
 - name: EC(e)
   composition: {C: 3, H: 4, O: 3}
   thermo: {model: constant-cp}
+  equation-of-state:
+    model: constant-volume
+    density: 2.26 g/cm^3
```

Comment:
Per the deprecation warning implemented in this PR, this should likely use `molar-density` (here and below).

```
    warn_deprecated("class LatticePhase", "To be removed after Cantera 3.2. Can be "
        "replaced by use of IdealSolidSolnPhase with the site density used to set the "
        "molar density of each constituent species.");
```

Reply:
The species `equation-of-state` field is flexible in terms of what input is provided for a constant volume species -- (mass) density, molar density, or molar volume are all supported. The deprecation warning suggests molar density as the most direct replacement for the "site" density that is the only allowed form in the `LatticePhase` class.